### PR TITLE
One workflow for align&qc and variant calling

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Changed: merged alignment, quality control, and variant calling workflows into one.

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by hassanfa** [comment]
Ready to review!

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule [hunk 1]
```diff
@@ -156,7 +156,7 @@ rule sentioen_filter_TNscope:
         tnscope_filtered_vcf = vcf_dir + "SNV.somatic." + config["analysis"]["case_id"] + ".tnscope.vcf.gz",
     params:
         tmpdir = tmp_dir,
-        sentieon_ml_tnscope = SENTIEON_TNSCOPE,
+        sentieon_ml_tnscope = config["SENTIEON_TNSCOPE"],
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule:159]: Is this variable written in config.json file? In my current ongoing PR, I made changes in `commands/config/case.py`, where the Sentieon_exec and sentieon_lic are defined.

File: BALSAMIC/workflows/balsamic.smk [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+SENTIEON_LICENSE = ''
+SENTIEON_INSTALL_DIR = ''
+
+# explicitly check if cluster_config dict has zero keys.
+if len(cluster_config.keys()) == 0:
+    cluster_config = config
+
+try:
+    config["SENTIEON_LICENSE"] = os.environ["SENTIEON_LICENSE"]
+    config["SENTIEON_INSTALL_DIR"] = os.environ["SENTIEON_INSTALL_DIR"]
```
Comment (ashwini06) [inline BALSAMIC/workflows/balsamic.smk:47]: 👍 I will also add these changes to my current PR :) At the moment, I predefined the `sentieon_dir` and `license` in `commands/config/case.py`

**Review by ashwini06 — COMMENTED** [review]
Looks good. Much simpler to go through balsamic rules in just one workflow:)

File: BALSAMIC/workflows/balsamic.smk (hunk 2 above)
Comment (hassanfa) [reply BALSAMIC/workflows/balsamic.smk:47]: I think it is better to handle these on runtime. Workflow run change according to what is available, and SLURM jobs won't fail if Sentieon license fails to connect or work.

**Comment by hassanfa** [comment]
@ashwini06 I'll go ahead and merge this.

UMI workflow needs a checkpoint result from main variant calling workflow for the noise reduction calculation. And that's a big decision that I'd like to take after discussing with you and Snakefile changes. Let me know if any of this breaks your PR.

**Comment by ashwini06** [comment]
> @ashwini06 I'll go ahead and merge this.
> 
> UMI workflow needs a checkpoint result from main variant calling workflow for the noise reduction calculation. And that's a big decision that I'd like to take after discussing with you and Snakefile changes. Let me know if any of this breaks your PR.

👍 Mostly these changes shouldn't interfere with UMI workflow.

**Pull request merged**